lib/checkcondition.cpp

```diff
@@ -1567,13 +1567,16 @@ void CheckCondition::checkDuplicateConditionalAssign()
             if (!isSameExpression(
                     mTokenizer->isCPP(), true, condTok->astOperand2(), assignTok->astOperand2(), mSettings->library, true, true))
                 continue;
-            duplicateConditionalAssignError(condTok);
+            duplicateConditionalAssignError(condTok, assignTok);
         }
     }
 }
 
-void CheckCondition::duplicateConditionalAssignError(const Token *tok)
+void CheckCondition::duplicateConditionalAssignError(const Token *condTok, const Token* assignTok)
 {
+    ErrorPath errors;
+    errors.emplace_back(assignTok, "Assignment to same expression in condition.");
```

Comment:
I suggest that the note for assignTok will say something like "Assign value 0 to x, even though x is already 0." and the note for condTok will say "Condition ensure that x is 0" ... something like that.

Reply:
> I suggest that the note for assignTok will say something like "Assign value 0 to x, even though x is already 0." 

I can do that for `if (x == y) x = y`, but how should I write the message for `if (x != y) x = y`?

> the note for condTok will say "Condition ensure that x is 0" ... something like that.

I dont follow this.

Reply:
In the case where we have:

    if (x == y) {
        x = y;
    }

I would suggest:

    errors.emplace_back(condTok, "Condition 'x==y'");
    errors.emplace_back(assignTok, "Assignment 'x=y' is redundant");

If the code is != :

    if (x != y)
        x = y;

Then maybe opposite notes would be more proper:

    errors.emplace_back(assignTok, "Assignment 'x=y'");
    errors.emplace_back(condTok, "Condition 'x!=y' is redundant");